django/db/backends/postgresql/features.py

```diff
@@ -74,17 +76,9 @@ def allows_group_by_selected_pks_on_model(self, model):
             return True
 
         try:
-            return not model._meta._is_view
+            return model._meta._is_ordinary_table
         except AttributeError:
+            intr = DatabaseIntrospection(self.connection)
             with self.connection.cursor() as cursor:
-                cursor.execute("""
-                    SELECT 1
-                    FROM pg_class c
-                        JOIN pg_namespace n ON c.relnamespace = n.oid
-                    WHERE c.relname = %s
-                        AND c.relkind = 'r'
-                        AND n.nspname NOT IN ('pg_catalog', 'pg_toast')
-                        AND pg_catalog.pg_table_is_visible(c.oid)
-                     """, (model._meta.db_table,))
-                model._meta._is_view = cursor.fetchone() is None
-            return not model._meta._is_view
+                model._meta._is_ordinary_table = intr.is_ordinary_table(cursor, model._meta.db_table)
```

Comment:
You can use `self.connection.introspection.is_ordinary_table(...)`.

Reply:
Didn't know that, thanks! I also see the introspection is a child of `BaseDatabaseIntrospection`, is it okay that the Posgres' introspection has the extra `is_ordinary_table` method that's not defined in `BaseDatabaseIntrospection`?

Reply:
Yes, that's fine.